model/riscv_mem.sail

```diff
@@ -142,7 +142,7 @@ function checked_mem_read forall 'n, 0 < 'n <= max_mem_access . (t : AccessType(
 
 /* PMP checks if enabled */
 function pmp_mem_read forall 'n, 0 < 'n <= max_mem_access . (t : AccessType(ext_access_type), p : Privilege, paddr : xlenbits, width : atom('n), aq : bool, rl : bool, res: bool, meta : bool) -> MemoryOpResult((bits(8 * 'n), mem_meta)) =
-  if   (~ (plat_enable_pmp ()))
+  if   (~ ((plat_enable_pmp ()) | (haveSmepmp())))
```

Comment:
Surely you can't have Smepmp without the base PMP? That is, surely haveSmepmp should imply plat_enable_pmp and thus be redundant?

Reply:
Yes, you are right. I did know that this check is redundant waiting for this [PR](https://github.com/riscv/sail-riscv/pull/246) to get merged then will push the final changes.